src/main/java/org/ays/common/util/validation/EmailValidator.java

```diff
@@ -0,0 +1,46 @@
+package org.ays.common.util.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+
+
+/**
+ * A custom validator implementation for the {@link Email} annotation.
+ * Validates whether the provided email invalid domain
+ * specified regular expression.
+ */
+
+class EmailValidator implements ConstraintValidator<Email, String> {
+
+    private static final String EMAIL_REGEX =
+            "^[a-zA-Z0-9]+([._-][a-zA-Z0-9]+)*@[a-zA-Z0-9]+" + "([.-]?[a-zA-Z0-9]+)+\\.[a-zA-Z]{2,}$";
+
+    /**
+     * Checks whether the given value is a valid email or not.
+     * <p>Some valid emails are:</p>
+     * <ul>
+     *     <li>user@example.com</li>
+     *     <li>john.doe123@example.co.uk</li>
+     *     <li>admin_123@example.org</li>
+     * </ul>
+     *
+     * <p>Some invalid emails are:</p>
+     * <ul>
+     *     <li>user@invalid</li>
+     *     <li>user@invalid!.com</li>
+     *     <li>user@.com</li>
+     * </ul>
+     *
+     * @param email object to validate
+     * @return true if the value is valid, false otherwise
+     */
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraintValidatorContext) {
+
+        if (!email.matches(EMAIL_REGEX)) {
+            return false;
+        }
+
+        return email.matches(EMAIL_REGEX);
```

Comment:
Aşağıdaki şekilde yazmamız yeterli olur. Zaten matches olmazsa false dönecek. Ek olarak burada email'in boş geldiği durumlarda regex ile karşılaştırmak yerine validasyondan geçmesini sağlamamız lazım çünkü öbür türlü `@NotBlank` anotasyonuyla çakışabilir.

```suggestion

        if (!String.hasText(email)) {
            return true;
        }

        return email.matches(EMAIL_REGEX);
```